contracts/governed_token_wrapper/lib.rs

```diff
@@ -0,0 +1,531 @@
+        #[ink(message, payable)]
+        pub fn wrap_for_and_send_to(
+            &mut self,
+            token_address: Option<AccountId>,
+            sender: AccountId,
+            amount: Balance,
+            recipient: AccountId,
+        ) {
+            self.is_valid_wrapping(token_address, amount);
+
+            // determine amount to use
+            let amount_to_use = if token_address.is_none() {
+                self.env().transferred_value()
+            } else {
+                amount
+            };
+
+            let cost_to_wrap = self.get_fee_from_amount(amount_to_use);
+
+            let leftover = amount_to_use.saturating_sub(cost_to_wrap);
+
+            self.do_wrap(
+                token_address.clone(),
+                sender,
+                recipient,
+                cost_to_wrap,
+                leftover,
+            );
+        }
+
+        /// Used to unwrap/burn the wrapper token on behalf of a sender.
+        ///
+        /// token_address is the address of PSP22 to transfer to, if token_address is None,
+        /// then it's a Native token address
+        ///
+        /// amount is the amount of token to transfer
+        ///
+        /// sender is the Address of sender where liquidity is send to.
+        #[ink(message, payable)]
+        pub fn unwrap_for(
+            &mut self,
+            token_address: Option<AccountId>,
+            amount: Balance,
+            sender: AccountId,
+        ) {
+            self.is_valid_unwrapping(token_address, amount);
+            self.do_unwrap(token_address.clone(), sender, sender, amount);
+        }
+
+        ///  Adds a token at `_tokenAddress` to the GovernedTokenWrapper's wrapping list
+        ///
+        /// tokenAddress:  The address of the token to be added
+        ///
+        /// nonce: The nonce tracking updates to this contract
+        #[ink(message)]
+        pub fn add_token_address(&mut self, token_address: AccountId, nonce: u64) -> Result<()> {
+            // only contract governor can execute this function
+            self.is_governor(self.env().caller());
+
+            // check if token address already exists
+            if self.is_valid_address(token_address) {
+                return Err(Error::TokenAddressAlreadyExists);
+            }
+
+            if self.proposal_nonce > nonce {
+                return Err(Error::InvalidNonce);
+            }
+
+            if nonce != self.proposal_nonce + 1 {
+                return Err(Error::NonceMustIncrementByOne);
+            }
+
+            self.valid.insert(token_address, &true);
+            self.historically_valid.insert(token_address, &true);
+            self.tokens.insert(token_address, &true);
+            self.historical_tokens.insert(token_address, &true);
+
+            self.proposal_nonce = nonce;
+
+            Ok(())
+        }
+
+        ///  Removes a token at `_tokenAddress` from the GovernedTokenWrapper's wrapping list
```

Comment:
Function comments look to have extra spaces. Please fix on all functions.

Reply:
Also the comment looks to be pointing to Solidity naming conventions and not to this contract. Please fix accordingly for all functions.